scipy/signal/signaltools.py

```diff
@@ -395,36 +469,334 @@ def fftconvolve(in1, in2, mode="full", axes=None):
+
+    This is generally much faster than `convolve` for large arrays (n > ~500),
+    and generally much faster than `fftconvolve` when one array is much
+    larger than the other, but can be slower when only a few output values are
+    needed or when the arrays are very similar in shape, and can only
+    output float arrays (int or object array inputs will be cast to float).
+
+    Parameters
+    ----------
+    in1 : array_like
+        First input.
+    in2 : array_like
+        Second input. Should have the same number of dimensions as `in1`.
+    mode : str {'full', 'valid', 'same'}, optional
+        A string indicating the size of the output:
+
+        ``full``
+           The output is the full discrete linear convolution
+           of the inputs. (Default)
+        ``valid``
+           The output consists only of those elements that do not
+           rely on the zero-padding. In 'valid' mode, either `in1` or `in2`
+           must be at least as large as the other in every dimension.
+        ``same``
+           The output is the same size as `in1`, centered
+           with respect to the 'full' output.
+           axis : tuple, optional
+    axes : int or array_like of ints or None, optional
+        Axes over which to compute the convolution.
+        The default is over all axes.
+
+
+    Returns
+    -------
+    out : array
+        An N-dimensional array containing a subset of the discrete linear
+        convolution of `in1` with `in2`.
+
+    See Also
+    --------
+    convolve : Uses the direct convolution or FFT convolution algorithm
+               depending on which is faster
+    fftconvolve : An implementation of convolution using FFT.
+
+    Notes
+    -----
+    The function will automatically fall back on using `fftconvolve` in
+    situations where the overlap-add method does not work.
+
+    .. versionadded:: 1.4.0
+
+    Examples
+    --------
+    Convolve a 100,000 sample signal with a 512-sample filter.
+
+    >>> from scipy import signal
+    >>> sig = np.random.randn(100000)
+    >>> filt = signal.firwin(512, 0.01)
+    >>> fsig = signal.oaconvolve(sig, filt)
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, (ax_orig, ax_mag) = plt.subplots(2, 1)
+    >>> ax_orig.plot(sig)
+    >>> ax_orig.set_title('White noise')
+    >>> ax_mag.plot(fsig)
+    >>> ax_mag.set_title('Filtered noise')
+    >>> fig.tight_layout()
+    >>> fig.show()
+
+    References
+    ----------
+    .. [1] Wikipedia, "Overlap-add_method".
+           https://en.wikipedia.org/wiki/Overlap-add_method
+
+    """
+    in1 = asarray(in1)
+    in2 = asarray(in2)
+
+    # NOTE the undocumented `_oa_only` argument, if set to `True`, will result
+    # in an exception if the function falls back on `fftconvolve`.  This is
+    # to guarantee that oaconvolve function itself, rather than the
+    # fftconvolve fallback mode, is being used in tests and benchmarks
+    oa_only_error = ValueError('Using fftconvolve with_oa_only==True')
```

Comment:
Do we really want to do this?  I'm not a big fan of a supposedly hidden argument that is actually exposed in the function definition and then therefore in the documentation.  Is there any precedence in Scipy for this kind of thing?

Reply:
I've replaced this with a normal argument.

Reply:
I have eliminated this entirely by using pytest's monkeypatch as discussed below.